### PR TITLE
Nested search preparation

### PR DESCRIPTION
**Pull request by gnusi**
Ensure document blocks are not splitted during consolidation

### CODE REVIEW

File: core/index/merge_writer.cpp [hunk 1]
```diff
@@ -1109,103 +1044,96 @@ std::optional<field_id> columnstore::insert(
...
+      } else if ((*less_)(lhs_value, rhs_value)) {
+        return false;
+      }
+
+      // tie braker to avoid splitting document blocks
+      if (lhs_it.it.get() == rhs_it.it.get()) {
+        return lhs_it.doc->value < rhs_it.doc->value;
+      }
+
+      return lhs_it.it.get() < rhs_it.it.get();
```
Comment (MBkkt) [inline core/index/merge_writer.cpp:1112]: Why not?
```suggestion
      return lhs_it.it.get() < rhs_it.it.get() || 
             (lhs_it.it.get() == rhs_it.it.get() && lhs_it.doc->value < rhs_it.doc->value);
```

**Review by MBkkt — APPROVED** [review]
LGTM

**Review by Dronplane — APPROVED** [review]
LGTM

File: core/index/merge_writer.cpp (hunk 1 above)
Comment (gnusi) [reply core/index/merge_writer.cpp:1112]: Isn't it the same? ;-)

**Pull request merged**